### DISCUSSION TITLE
Unhandeled exception

### DISCUSSION THREAD

**Discussion by HectorServadac** [Rem0o/FanControl.Releases#670] [General]
Hi RemOo,

Just started using FanControl and I have to admit you have created an incredible piece of software! So far everything is fine for the most part.

Yesterday I started receiving an unhandeled exception that would end up in the disappearance of the rightest curve of the section. At first, I thought it was crashing at random, for no apparent reason, but I finally understood the reason when I wanted to rearrange the order in which my curves were positioned. 

It happens every time you click on the empty space at the right of the rightest curve in the panel and move the cursor. I think the program expect a curve to be there and tries to move it although there isn't, and for some reason, it removes the curve that is on the left of the space.  Not that big of a deal, you just have to reload your save config, but it still a bug, so I wanted to let you know.

Another thing that could be modified is that there is no "cancel" option on the Curve editor. I would be great to be able to just ignore/cancel any modification made to the curve.

Here is the log file infos for the exception:

2022-04-25 08:51:31: Unhandled exception in FanControl v1.0.0.0
2022-04-25 08:51:31: System.ArgumentOutOfRangeException: L'index doit être dans les limites de la List.
Nom du paramètre : index
   à System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   à System.Collections.ObjectModel.Collection`1.Insert(Int32 index, T item)
   à System.Collections.ObjectModel.Collection`1.System.Collections.IList.Insert(Int32 index, Object value)
   à GongSolutions.Wpf.DragDrop.DefaultDropHandler.Drop(IDropInfo dropInfo)
   à GongSolutions.Wpf.DragDrop.DragDrop.DropTargetOnDrop(Object sender, DragEventArgs e, EventType eventType)
   à System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   à System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   à System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   à System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   à System.Windows.OleDropTarget.RaiseDragEvent(RoutedEvent dragEvent, Int32 dragDropKeyStates, Int32& effects, DependencyObject target, Point targetPoint)
   à System.Windows.OleDropTarget.MS.Win32.UnsafeNativeMethods.IOleDropTarget.OleDrop(Object data, Int32 dragDropKeyStates, Int64 point, Int32& effects)
   à MS.Win32.UnsafeNativeMethods.DoDragDrop(IDataObject dataObject, IOleDropSource dropSource, Int32 allowedEffects, Int32[] finalEffect)
   à System.Windows.OleServicesContext.OleDoDragDrop(IDataObject dataObject, IOleDropSource dropSource, Int32 allowedEffects, Int32[] finalEffect)
   à System.Windows.DragDrop.OleDoDragDrop(DependencyObject dragSource, DataObject dataObject, DragDropEffects allowedEffects)
   à System.Windows.DragDrop.DoDragDrop(DependencyObject dragSource, Object data, DragDropEffects allowedEffects)
   à GongSolutions.Wpf.DragDrop.DragDrop.DoDragSourceMove(Object sender, Func`2 getPosition)
   à GongSolutions.Wpf.DragDrop.DragDrop.DragSourceOnMouseMove(Object sender, MouseEventArgs e)
   à System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   à System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   à System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   à System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   à System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   à System.Windows.Input.InputManager.ProcessStagingArea()
   à System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   à System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   à System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   à System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   à System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   à MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   à MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   à System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   à System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Thanks.

**Comment by Rem0o** [comment]
Yup I know, it got reported already, will be fixed in the next release.

**Comment by HectorServadac** [comment]
Ohhh, great didn't find it reported in the forum so. I hope you read my suggestion about curve editor cancel button too. 

Thanks.

**Reply by Rem0o** [reply]
Well you kinda can if you X out of the software and discard your changes, but I don't hate the idea of a cancel button there.
